Problem_3.js

```diff
@@ -0,0 +1,52 @@
+https://leetcode.com/problems/spiral-matrix/
+// Time Complexity : O(m * n)
+// Space Complexity : O(m * n) result array
```

Comment:
Result array is the output array , output space does not count in space complexity